spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java

```diff
@@ -220,6 +226,7 @@ public static WriteMethod from(@Nullable String writeMethod) {
   private long readSessionCacheDurationMins = 5L;
   private SparkBigQueryProxyAndHttpConfig sparkBigQueryProxyAndHttpConfig;
   private CompressionCodec arrowCompressionCodec = DEFAULT_ARROW_COMPRESSION_CODEC;
+  private ResponseCompressionCodec responseCompressionCodec = DEFAULT_RESPONSE_COMPRESSION_CODEC;
```

Comment:
SparkBigQueryConfig is a Serializable class, and this enum is not Serializable. The field should be a `String`, converted to a ResponseCompressionCodec only on the executor.

Reply:
I'm not sure I understand why CompressionCodec is an serializable enum, but ResponseCompressionCodec is not a serializable enum.

Reply:
Looking at `ResponseCompressionCodec` inheritance tree I cannot see Serializable and therefore we need to convert it to another type (int/String). Alternatively, add a test to SparkBigQueryConfigTest verifying it it still Serializable

Reply:
I already have  testDefaults()  in SparkBigQueryConfigTest.  is that insufficient?